### PR TITLE
Redo distance calculation & add unit tests for it

### LINKED ISSUE #148: Race Optimization Distance Calculation

### PR DESCRIPTION
**Pull request by bill-luu**
The original assumption that an odometer can't decrease in value was wrong :( , so it caused massive discrepancies. We were also not rounding the values from meters to kilometers.

I also added some chai unit tests, that can be run by installing mocha and running `./node_modules/mocha/bin/mocha` in the server directory.

~I'll do a future PR for documentation & getting travis to run it the unit tests~ You should be able to run the tests with `npm run test`

Fixes #148

### CODE REVIEW

File: server/test/app.js [hunk 1]
```diff
@@ -1,11 +1,48 @@
...
+      // Arrays should be sorted newest first
+      const packetArray = [
+        {'motor0odometer': 10000, 'motor1odometer': 1000},
+        {'motor0odometer': 5000, 'motor1odometer': 2000},
+        {'motor0odometer': 5000, 'motor1odometer': 0},
+      ];
+      assert.equal(3.0, race.getDistanceTraveled(packetArray));
+    });
 
-describe('GET /random-url', () => {
```
Comment (LiamConway) [inline server/test/app.js]: Just want to ensure removing this test case was intentional

Comment (bill-luu) [reply server/test/app.js]: Yes, I'm not sure if it works lol

**Comment by bill-luu** [comment]
Is there any places I can add an extra comment on so that it's more clearer what the calculation is supposed to do?

**Comment by LiamConway** [comment]
After speaking with @bill-luu in Slack, calculations look good :+1:

**Pull request merged**